build-lib.js

```diff
@@ -0,0 +1,69 @@
+const fs = require("fs");
+const path = require("path");
+
+// Paths
+const distPath = path.join(__dirname, "dist");
+const jsFilePath = path.join(distPath, "index.js");
+const dtsFilePath = path.join(distPath, "index.d.ts");
+const modulePath = path.join(distPath, "tutorial-module", "index.js");
+
+// File contents
+const jsContent = 'export { TutorialModule } from "./tutorial-module/index";';
+const dtsContent = 'export { TutorialModule } from "./tutorial-module/TutorialRoot";';
+
+function validateDistFolder() {
+    if (!fs.existsSync(distPath)) {
+        fs.mkdirSync(distPath);
+        console.log("'dist' directory created.");
+    } else {
+        console.log("'dist' directory already exists.");
+    }
+}
+
+// Generate the files
+function generateIndexJsWithTypes() {
+    // Check if ./tutorial-module/index.js exists
+    if (fs.existsSync(modulePath)) {
+        console.log("The './tutorial-module/index.js' file exists. Generating files...");
+        // Write index.js
+        fs.writeFile(jsFilePath, jsContent, err => {
+            if (err) {
+                console.error(`Error generating 'index.js': ${err.message}`);
+            } else {
+                console.log("'index.js' generated successfully.");
+            }
+        });
```

Comment:
Instead of the old callback style, use promises  "import { promises as fs } from 'fs';"  or simply the "Sync" methods (like you do in exists)